File: leetCode/fukun/201-300/lc_283.js
// 示例:
//
//     输入: [0,1,0,3,12]
// 输出: [1,3,12,0,0]
// 说明:
//
//     必须在原数组上操作，不能拷贝额外的数组。
// 尽量减少操作次数。
//
// 来源：力扣（LeetCode）
// 链接：https://leetcode-cn.com/problems/move-zeroes
//     著作权归领扣网络所有。商业转载请联系官方授权，非商业转载请注明出处。
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    let curr = 0;
    for (let i = 0; i < nums.length; i++) nums[i] !== 0 ? nums[curr++] = nums[i] : true;
    while (curr < nums.length) nums[curr++] = 0;
    return nums;
};

console.log(moveZeroes([0, 0, 1]));

File: leetCode/XSS/lc_283.java
public class lc_283 {
    //移动零
    public static void main(String[] args) {

    }
    static  class Solution {
        public void moveZeroes(int[] nums) {
            int k=0;
            for(int i=0;i<nums.length;i++)
            {
                if(nums[i]==0)
                {
                    k++;
                }
                else
                {
                    nums[i-k]=nums[i];
                }
            }
            for(int j=nums.length-k;j<nums.length;j++)
            {
                nums[j]=0;
            }
        }
    }
}
